zcash_primitives/src/sapling/builder.rs

```diff
@@ -752,11 +742,7 @@ impl<P: InProgressProofs, V> Bundle<InProgress<P, Unsigned>, V> {
             MaybeSigned::SigningMetadata,
             |auth: InProgress<P, Unsigned>| InProgress {
                 sigs: PartiallyAuthorized {
-                    binding_signature: auth.sigs.bsk.sign(
-                        &sighash,
-                        &mut rng,
-                        VALUE_COMMITMENT_RANDOMNESS_GENERATOR,
-                    ),
+                    binding_signature: auth.sigs.bsk.sign(&mut rng, &sighash),
```

Comment:
There was a bug here previously in that this didn't include key prefixing, because we never implemented that inside our `redjubjub` module (as the original RedDSA draft spec lacked it). I should have used the `sapling::spend_sig` method here when I added this in #1023, but as we haven't cut a release including this API yet, the bug is not in deployed code. The `redjubjub` crate includes key prefixing.